rxbroadcast/src/main/java/com/cantrowitz/rxbroadcast/BroadcastProvider.java

```diff
@@ -32,29 +31,28 @@ public void call(final Subscriber<? super Intent> subscriber) {
         final Subscription subscription = Subscriptions.create(new Action0() {
             @Override
             public void call() {
-                unregisterBroadcastReceiver(broadcastReceiver);
+                broadcastRegistrarStrategy.unregisterBroadcastReceiver(broadcastReceiver);
             }
         });
 
-        registerBroadcastReceiver(broadcastReceiver);
+        broadcastRegistrarStrategy.registerBroadcastReceiver(broadcastReceiver);
         subscriber.add(subscription);
     }
 
     @VisibleForTesting
-    BroadcastReceiver getBroadcastReceiver(){
+    BroadcastReceiver getBroadcastReceiver() {
         return broadcastReceiver;
     }
 
-    protected abstract void registerBroadcastReceiver(BroadcastReceiver broadcastReceiver);
-
-    protected abstract void unregisterBroadcastReceiver(BroadcastReceiver broadcastReceiver);
 
     @NonNull
     private BroadcastReceiver createBroadcastReceiver(final Subscriber<? super Intent> subscriber) {
```

Comment:
I try and have my methods do one thing. I find that's it's easier to read and maintain. I think this is largely personal and has no impact on the function of this area. 

"One way to know that a function is doing more than 'one thing' is if you can extract another function from it with a name that is not merely a restatement of its implementation." -- Clean Code by Robert C. Martin

If it is more of the standard of RX to do it inline, I'd be happy to do that as well.